### PR TITLE
Implements basic experiment to measure white-box feature performance

### PR DESCRIPTION
**Pull request by vulder**

### CODE REVIEW

File: varats-core/varats/project/project_util.py [hunk 1]
```diff
@@ -163,12 +163,23 @@ class ProjectBinaryWrapper():
...
+        binary_type: BinaryType,
+        entry_point: tp.Optional[Path] = None
     ) -> None:
         self.__binary_name = binary_name
         self.__binary_path = path_to_binary
         self.__type = binary_type
 
+        if binary_type is BinaryType.EXECUTABLE:
+            self.__entry_point = entry_point
+            if not self.entry_point:
```
Comment (jonas-kaufmann) [inline varats-core/varats/project/project_util.py:235]:
```suggestion
            if not self.__entry_point:
```

Typo?

File: varats-core/varats/report/tef_report.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    ASYNC_EVENT_INSTANT = 'n'
+    ASYNC_EVENT_END = 'e'
+    FLOW_EVENT_START = 's'
+    FLOW_EVENT_STEP = 't'
+    FLOW_EVENT_END = 'f'
+    SAMPLE_EVENT = 'P'
+
+    @staticmethod
+    def parse_event_type(raw_event_type: str) -> 'TraceEventType':
+        """Parses a raw string that represents a trace-format even type and
```
Comment (jonas-kaufmann) [inline varats-core/varats/report/tef_report.py]:
```suggestion
        """Parses a raw string that represents a trace-format event type and
```

File: varats-core/varats/utils/git_util.py [hunk 3]
```diff
@@ -773,22 +773,26 @@ def specify_binary(
...
         Args:
-            location: where the binary can be found, relative to the \
+            location: where the binary can be found, relative to the
                       project-source root
             binary_type: the type of binary that is produced
             override_binary_name: overrides the used binary name
-            only_valid_in: additinally specifies a validity range that \
-                           specifies in which revision range this binary is \
+            override_entry_point: overrides the executable entry point
+            only_valid_in: additinally specifies a validity range that
```
Comment (jonas-kaufmann) [inline varats-core/varats/utils/git_util.py]:
```suggestion
            only_valid_in: additionally specifies a validity range that
```

File: tests/utils/test_project_util.py [hunk 4]
```diff
@@ -226,3 +228,27 @@ def test_if_project_names_are_well_formed(self) -> None:
                     f"The project name {project_name} must not contain the "
                     f"dash character."
                 )
+
+
+class TestProjectBinaryWrapper(unittest.TestCase):
+    """Test if we can correctly setup and use the RevisionBinaryMap."""
+
+    def test_execution_of_executable(self) -> None:
+        """Check if we can execute a executable bianries."""
```
Comment (jonas-kaufmann) [inline tests/utils/test_project_util.py]:
```suggestion
        """Check if we can execute an executable binary."""
```

File: tests/utils/test_project_util.py [hunk 5]
```diff
@@ -226,3 +228,27 @@ def test_if_project_names_are_well_formed(self) -> None:
...
+        binary = ProjectBinaryWrapper("ls", "/bin/ls", BinaryType.EXECUTABLE)
+
+        ret = binary()
+        self.assertIsNotNone(ret)
+        self.assertIsInstance(ret, str)
+
+    def test_execution_of_libraries(self) -> None:
+        """Check if we don't fail when executing a shared/static library."""
```
Comment (jonas-kaufmann) [inline tests/utils/test_project_util.py]:
```suggestion
        """Check whether we fail when executing a shared/static library."""
```

**Review by jonas-kaufmann — COMMENTED** [review]
Found some typos.

File: varats-core/varats/project/project_util.py (hunk 1 above)
Comment (vulder) [reply varats-core/varats/project/project_util.py:235]: nope, this one is ok
The class defines `entry_point` as a property, so we call here the "getter", see line 200

File: varats-core/varats/report/tef_report.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        """Parses a raw string that represents a trace-format event type and
+        converts it to the corresponding enum value."""
+        for trace_event_type in TraceEventType:
+            if trace_event_type.value == raw_event_type:
+                return trace_event_type
+
+        raise LookupError("Could not find correct trace event type")
+
+    def __str__(self) -> str:
+        return self.value[0]
```
Comment (vulder) [inline varats-core/varats/report/tef_report.py]:
```suggestion
        return self.value
```

**Review by boehmseb — APPROVED** [review]

**Pull request merged**